### ISSUE TITLE
Need add support of retina displays

### ISSUE THREAD

**Issue by ioannismihailidis** [RSATom/QuickLayer#1]
same problem in QQuickRenderControl example with qt 5.4 under mac os x 10.10.2. Which Os are you using?

**Comment by RSATom** [comment]
I am on OS X 10.10.2 too. I've checked demo - it works fine for me. What hardware do you have?

**Comment by ioannismihailidis** [comment]
MBP Retina 2012, Nvidia Geforce 650M, Qt5.4 downloaded with online MaintenanceTool. Any idea what might be the issue?

**Comment by RSATom** [comment]
I didn't try it on retina display, and I have only integrated video (Intel hd 4000). Maybe it's related.
Do you see anything in debug output?

**Comment by ioannismihailidis** [comment]
Intergrated has the same problem. BUT: Running the app in low (non retina) resolution fixes the problem!  Thanks!

**Comment by RSATom** [comment]
It's very interesting and will be very useful to me too. So thank you! :)
Could you please share solution if you will find it?

**Comment by ioannismihailidis** [comment]
I will let you know! In the meanwhile I will wait for qt5.5...maybe the problem will be solved  :)

**Comment by RSATom** [comment]
I can't wait 5.5 :) So maybe will have to fix it myself. Will write here if will find solution.

**Comment by jherico** [comment]
I suggest you check https://bugreports.qt.io/browse/QTBUG-45613 and https://bugreports.qt.io/browse/QTBUG-42114 (the latter is reported fixed, but I'm having this problem, so I cloned it as the former).

I added a significant amount of detail to QTBUG-45613 after capturing the GL calls with a tracing program.  The problem seems to be a bad glViewport call (at least) when on the retina display.

**Comment by RSATom** [comment]
@jherico, thank you for info.
Do you use ~~QmlVlc~~ QuickLayer? I just don't have access to Mac with retina display and thus can't test any solution.

**Comment by jherico** [comment]
no, I don't, sorry.  I may have to dive into the code myself, but for the time being I'm disabling the HDPI functionality on my retina display.

**Comment by RSATom** [comment]
If problem related to glViewport, then it could be fixed just by adding at https://github.com/RSATom/QuickLayer/blob/master/QuickLayer.mm#L115 something like this:

```
    glViewport(0, 0, fbo->width(), fbo->height());
```
